azurerm/resource_arm_application_gateway.go

```diff
@@ -1560,16 +1636,86 @@ func flattenApplicationGatewayConnectionDraining(input *network.ApplicationGatew
 }
 
 func expandApplicationGatewaySslPolicy(d *schema.ResourceData) *network.ApplicationGatewaySslPolicy {
-	vs := d.Get("disabled_ssl_protocols").([]interface{})
-	results := make([]network.ApplicationGatewaySslProtocol, 0)
+	policy := network.ApplicationGatewaySslPolicy{}
+	vs := d.Get("ssl_policy").([]interface{})
+	if len(vs) == 0 {
+		return &policy
```

Comment:
if `ssl_policy` isn't specified but `disabled_ssl_protocols` is (e.g. upgrading with no code changes) - at the moment that'll cause none to be disabled (since it's read below https://github.com/terraform-providers/terraform-provider-azurerm/pull/3360/files#diff-b945a735bb28605a8bcb81914018d807R1650) - can we parse out `disabled_ssl_protocols` here?